src/components/changePasswordForm.jsx

```diff
@@ -0,0 +1,109 @@
+import "./../assets/css/changePassword.css";
+import { useState } from 'react';
+import { useFormik } from 'formik';
+import * as Yup from 'yup';
+import axios from "axios";
+
+
+const validationSchema = Yup.object().shape({
+    email: Yup.string().email('Invalid email').required('No email provided'),
+
+    oldPassword: Yup.string()
+    .min(8, 'Password must be at least 8 characters')
+    .required('No old password provided')
+    .matches(/^(?=.*[a-z])(?=.*[A-Z])(?=.*[!@#$%^&*(),.?":{}|<>]).*$/, 'Password should contain at least one letter, one special character, and one capital letter.'),
+
+    newPassword: Yup.string()
+    .min(8, 'Password must be at least 8 characters')
+    .required('No new password provided')
+    .matches(/^(?=.*[a-z])(?=.*[A-Z])(?=.*[!@#$%^&*(),.?":{}|<>]).*$/, 'Password should contain at least one letter, one special character, and one capital letter.'),
+  });
+  
+  const ChangePasswordPage = () => {
+    const formik = useFormik({
+      initialValues: {
+        email: '',
+        oldPassword: '',
+        newPassword: '',
+      },
+      validationSchema: validationSchema,
+      onSubmit:async (values) => {
+        try{
+            const res = await axios.put("https://e-comm-team-amigos-bn-project.onrender.com/user/createPassword", {email:values.email,oldpassword:values.oldPassword,newpassword:values.newPassword});
+            // localStorage.setItem("token",res.data.token);
+            console.log("success",res.data);
+            updateErrorDisplay.innerHTML = "Password updated successfuly!";
+            updateErrorDisplay.style.color = "#096F3E";
+            useEffect(()=>{
+              if(updatingPasswordSuccess)
+              toast(updatingPasswordSuccess)
+              if(updatingPasswordError)
+              toast(updatingPasswordError)
+              if(updatingPasswordSuccess === 'password updated successfully'){
+                  setTimeout(()=>{
+                      navigation("https://e-comm-team-amigos-bn-project.onrender.com/user/login")
+                  },5000)
+              }
+          },[updatingPasswordSuccess, updatingPasswordError])
+            console.log(res);
+        }catch(error){
+            if (error.response.status == 401) {
+              updateErrorDisplay.innerHTML = "Incorect email or old password!";
+              updateErrorDisplay.style.color = "red";
+            }else if (error.response.status == 404) {
+                updateErrorDisplay.innerHTML = "Email not found!";
+                updateErrorDisplay.style.color = "red";
+            }else {
+                console.log(error);
+            }
+        }
+    },
+  });
```

Comment:
Great Job for making integration with API but I'm suggesting to separate Formik validation and integration functionality. Further more, take advantage of using redux to manage states  in your component.